pgx-pg-sys/src/submodules/panic.rs

```diff
@@ -479,7 +506,15 @@ fn do_ereport(ereport: ErrorReportWithLevel) {
 
                 let sqlerrcode = ereport.sql_error_code();
                 let message = ereport.message().as_pg_cstr();
-                let detail = ereport.detail().as_pg_cstr();
+                let backtrace = ereport.backtrace();
+                let detail = ereport.detail();
+                let detail = match (detail, backtrace) {
```

Comment:
My only nit is that I'd prefer this logic be tucked away inside `ErrorReport::detail()` or a new method `ErrorReport::detail_with_backtrace()` or something similar.

Reply:
geez.  "this logic" means that match statement for `(detail, backtrace)`.